src/compas_fab/robots/robot.py

```diff
@@ -1156,6 +1168,15 @@ def inverse_kinematics(self, frame_WCF, start_configuration=None, group=None, re
         >>> robot.inverse_kinematics(frame_WCF, start_configuration, group)                 # doctest: +SKIP
         Configuration((4.045, 5.130, -2.174, -6.098, -5.616, 6.283), (0, 0, 0, 0, 0, 0))    # doctest: +SKIP
         """
+        # Pseudo-memoized sequential calls will re-use iterator if not exhaused
+        request_id = '{}-{}-{}-{}-{}'.format(id(frame_WCF), id(start_configuration), id(group), id(return_full_configuration), id(options))
```

Comment:
actually, i was just looking into `id` and the docs say "This is an integer which is guaranteed to be unique and constant for this object during its lifetime." So if an attribute of the frame changes or the value of some pair in the options change, then the `id` doesn't change.   Yay mutability! Do we want that?

Reply:
oh, and this is probably a one in a million bug, but 'Two objects with non-overlapping lifetimes may have the same id() value.'

Reply:
Oh my! good catch! especially the mutability part of the dict. I'll go for `str()` instead of `id()`